gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsException.java

```diff
@@ -16,17 +16,26 @@
 
 package com.google.gcloud.dns;
 
+import com.google.common.collect.ImmutableSet;
 import com.google.gcloud.BaseServiceException;
 import com.google.gcloud.RetryHelper.RetryHelperException;
 import com.google.gcloud.RetryHelper.RetryInterruptedException;
 
 import java.io.IOException;
+import java.util.Set;
 
 /**
  * DNS service exception.
  */
 public class DnsException extends BaseServiceException {
 
+  // see: https://cloud.google.com/dns/troubleshooting
+  private static final Set<Error> RETRYABLE_ERRORS = ImmutableSet.of(
+      new Error(404, null),
```

Comment:
Per discussion, we probably don't want to include 404 and 409.
Also, it would be good to involve the DNS people in this PR.